### PR TITLE
Partially address https://github.com/dotnet/roslyn-analyzers/issues/8…

### PR DESCRIPTION
**Pull request by mavasani**
…99: CA1815 doesn't match FxCop implementation

1. Do not fire for enums.
2. Do not fire for enumerators (needs IsAssignableTo helper that Larry is soon going to add).
3. Do not fire for value types without members.

### CODE REVIEW

**Comment by mavasani** [comment]
Tagging @dotnet/roslyn-analysis for review

File: src/Microsoft.ApiDesignGuidelines.Analyzers/Core/OverrideEqualsAndOperatorEqualsOnValueTypes.cs [hunk 1]
```diff
@@ -49,13 +50,42 @@ public sealed class OverrideEqualsAndOperatorEqualsOnValueTypesAnalyzer : Diagno
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(EqualsRule, OpEqualityRule);
 
+        protected virtual bool IsAssignableTo(INamedTypeSymbol type, INamedTypeSymbol assignableToType)
+        {
+            // TODO: Use the language specific helper for IsAssignableTo.
```
Comment (mavasani) [inline src/Microsoft.ApiDesignGuidelines.Analyzers/Core/OverrideEqualsAndOperatorEqualsOnValueTypes.cs]: This needs to invoke the new `IsAssignableTo` helper that @lgolding plans to add soon.

**Comment by natidea** [comment]
:+1:

**Comment by mavasani** [comment]
retest this please

**Comment by mavasani** [comment]
Debug build is failing due to an unrelated infrastructure issue - tests are passing locally, so merging it in.

**Comment by davkean** [comment]
@mavasani Can you file a bug on the infrastructure issue? It looks like something is picking up a binary from the output directory at the same time that someone is writing to it, or copying it.

**Comment by mavasani** [comment]
Sure, I am also going to try retesting it afterwards.

**Comment by mavasani** [comment]
@dotnet-bot retest Windows Debug please
// Previous failure: http://dotnet-ci.cloudapp.net/job/dotnet_roslyn-analyzers/job/master/job/windows_debug_prtest/46/
// Retest reason: Issue https://github.com/dotnet/roslyn-analyzers/issues/903

**Pull request merged**